### PR TITLE
if trigger disabled then send no event of problems

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

File: zabbix/stackstate_checks/zabbix/zabbix.py [hunk 1]
```diff
@@ -288,25 +288,34 @@ def retrieve_problems(self, url, auth):
...
+            severity = item.get("severity", priority)
 
             zabbix_problem = ZabbixProblem(event_id, acknowledged, trigger_id, severity)
 
             self.log.debug("Parsed ZabbixProblem %s." % zabbix_problem)
             if not event_id or not trigger_id or not severity:
                 self.log.warn("Incomplete ZabbixProblem, got: %s" % zabbix_problem)
 
-            yield zabbix_problem
+            if priority is not None:
```
Comment (whazenberg) [inline zabbix/stackstate_checks/zabbix/zabbix.py]: What is the case when there is no priority available? Add a comment and/or unit test?

File: zabbix/stackstate_checks/zabbix/zabbix.py [hunk 2]
```diff
@@ -288,25 +288,34 @@ def retrieve_problems(self, url, auth):
...
+                yield zabbix_problem
 
     def get_trigger_priority(self, url, auth, trigger_id):
+        # `monitored` flag make sure we only get enabled triggers
```
Comment (whazenberg) [inline zabbix/stackstate_checks/zabbix/zabbix.py]: Question; does that also include disabled triggers, the other support case?

File: zabbix/stackstate_checks/zabbix/zabbix.py [hunk 3]
```diff
@@ -288,25 +288,34 @@ def retrieve_problems(self, url, auth):
...
         params = {
             "output": ["priority"],
-            "triggerids": [trigger_id]
+            "triggerids": [trigger_id],
+            "monitored": True
```
Comment (whazenberg) [inline zabbix/stackstate_checks/zabbix/zabbix.py]: Unit test? for disabled and enabled triggers.

File: zabbix/stackstate_checks/zabbix/zabbix.py (hunk 2 above)
Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/zabbix.py]: No, we will get only enabled triggers and this is what we want from the scenario.

File: zabbix/stackstate_checks/zabbix/zabbix.py (hunk 1 above)
Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/zabbix.py]: if the trigger is enabled then for sure we will get the priority.

File: zabbix/stackstate_checks/zabbix/zabbix.py (hunk 3 above)
Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/zabbix.py]: test case added.

File: zabbix/stackstate_checks/zabbix/zabbix.py (hunk 1 above)
Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/zabbix.py]: put the comment for more clarification.

File: zabbix/tests/test_zabbix.py [hunk 4]
```diff
@@ -493,3 +493,31 @@ def test_zabbix_respect_true_ssl_verify(self):
...
+        def _mocked_method_request(url, name, auth=None, params={}, request_id=1):
+            if name == "apiinfo.version":
+                return self._apiinfo_response()
+            elif name == "host.get":
+                return self._zabbix_host_response()
+            elif name == "problem.get":
+                response = self._zabbix_problem()
+                return response
+            elif name == "trigger.get":
+                return None
```
Comment (whazenberg) [inline zabbix/tests/test_zabbix.py]: Does this work properly? Given that the code is:

```
response = self.method_request(url, "trigger.get", auth=auth, params=params)
trigger = response.get('result')
```

I'm expecting the response to be `response['result'] = []`

Comment (vishalxebia) [reply zabbix/tests/test_zabbix.py]: @whazenberg  correct I was wrong in this. I have made the change and also there should be an event for a host without any problems/triggers which will make the host healthy.

**Pull request merged**